core/src/main/java/net/sourceforge/jnlp/cache/ResourceDownloader.java

```diff
@@ -77,17 +77,46 @@ static void startDownload(Resource resource, final Object lock) {
         }
 
         if (!isProcessing) {
-            CachedDaemonThreadPoolProvider.DAEMON_THREAD_POOL.execute(new ResourceDownloader(resource, lock));
+            final ResourceDownloader rd = new ResourceDownloader(resource, lock);
+            CachedDaemonThreadPoolProvider.DAEMON_THREAD_POOL.execute(
+                new Runnable() {
+                    @Override
+                    public void run() {
+                        rd.runInitialize();
+                        CachedDaemonThreadPoolProvider.DAEMON_THREAD_POOL.execute(
```

Comment:
Why is the download done in a separate runnable?

Reply:
I like that the initialization is done in a separate thread from the actual download.
But the code needs to be cleaned up.
@hendrikebbers if it is OK for you we can accept the PR and do the code cleanup afterwards

Reply:
Fine with that. Will check if I have some time later today or tomorrow. In that case I will merge the PR and directly do a new one for the CleanUp.